app/src/test/groovy/pl/futurecollars/invoicing/controller/InvoiceControllerTest.groovy

```diff
@@ -0,0 +1,146 @@
+package pl.futurecollars.invoicing.controller
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import pl.futurecollars.invoicing.db.InMemoryDataBase
+import pl.futurecollars.invoicing.model.Invoice
+import pl.futurecollars.invoicing.utils.JsonService
+import spock.lang.Specification
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import static pl.futurecollars.invoicing.TestHelpers.invoice
+
+@AutoConfigureMockMvc
+@SpringBootTest
+class InvoiceControllerTest extends Specification{
+
+    @Autowired
+    private MockMvc mockMvc
+    @Autowired
+    private JsonService jsonService
+    @Autowired
+    private InMemoryDataBase inMemoryDataBase
+
```

Comment:
Jak chodzi o github actions, to spróbuj dodać metodę cleanup, podobnie jak przy testach z filebasedatabase, która będzie czyścić bazę między testami. Dodatkowo zadbaj proszę o formatowanie, bo widzę parę miejsc do poprawy :p